torchtune/modules/attention.py

```diff
@@ -176,8 +176,9 @@ def forward(
     ) -> torch.Tensor:
         """
         Args:
-            x (torch.Tensor): input tensor with shape [b x s_x x d]
-            y (Optional[torch.Tensor]): second input tensor for cross attention with shape [b x s_y x d]
+            x (torch.Tensor): input tensor with shape [b x s_x x d] for the query
+            y (Optional[torch.Tensor]): second input tensor for key and value with shape [b x s_y x d].
+                Optional only with kv_cache enabled.
```

Comment:
nit: this wording is confusing. Also can we explicit say something like "for self-attention, set y=x" somewhere?